### PR TITLE
Bump knip to v5

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Required for https://github.com/pixiebrix/pixiebrix-extension/pull/7941

## Team Coordination

- [ ] This PR requires an unlisted dependency update: let the team know on #engineering

## Checklist

- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -164,17 +164,12 @@ jobs:
...
-      - uses: actions/setup-node@v4
-        with:
-          node-version-file: package.json
-          cache: npm
-      - run: npm ci
       # https://knip.dev/overview/getting-started#without-installation
-      - run: npm install -D knip@4
-      # Do a run without an exit code. Knip's "warn" setting is not working for files and devDependencies.
-      - run: npm run dead-code:base -- --no-exit-code
-      # Automatically excludes exports tagged with `@knip` in their JSDoc
```
Comment (fregante) [inline .github/workflows/ci.yml:176]: - Info moved to config file
- --no-exit-code issue fixed

File: knip.mjs [hunk 2]
```diff
@@ -22,9 +22,6 @@ const knipConfig = {
     "src/background/messenger/external/api.ts",
     "src/store/browserExtensionIdStorage.ts",
     // Jest setup files
-    "src/testUtils/testEnv.js",
-    "src/testUtils/testAfterEnv.ts",
-    "src/testUtils/injectRegistries.ts",
```
Comment (fregante) [inline knip.mjs:27]: Dropped some outdated config

File: knip.mjs [hunk 3]
```diff
@@ -80,25 +71,14 @@ const knipConfig = {
...
-    // Incrementally enforce rules over time
-    exports: "error",
-    nsExports: "error",
-    files: "error",
-    duplicates: "error",
-    dependencies: "error",
-    enumMembers: "error",
-    classMembers: "error",
-    binaries: "error",
-    nsTypes: "error",
```
Comment (fregante) [inline knip.mjs:100]: Redundant config, already set via `--include` or by default

File: src/testUtils/renderWithCommonStore.ts [hunk 4]
```diff
@@ -54,6 +54,8 @@ const renderHookWithWrappers = createRenderHookWithWrappers(
 
 // eslint-disable-next-line import/export -- re-export RTL
 export * from "@testing-library/react";
+
+/** @knip Keep for consistency of API with other render helper modules */
```
Comment (fregante) [inline src/testUtils/renderWithCommonStore.ts:58]: Excludes via tag instead of via config

File: src/vendors/page-metadata-parser/parser.js [hunk 5]
```diff
@@ -229,8 +229,5 @@ function getMetadata(doc, url, customRuleSets) {
 }
 
 module.exports = {
-  buildRuleSet,
   getMetadata,
-  getProvider,
-  metadataRuleSets,
```
Comment (fregante) [inline src/vendors/page-metadata-parser/parser.js:235]: Fixes the issue reported instead of ignoring the whole file

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**